### ISSUE TITLE
[Bug] No Dice Animation on Crit/Fumble ("Dice So Nice"-Module)

### ISSUE THREAD

**Issue by ortegamarcel** [AnthonyMonette/TheWitcherTRPG#302]
## Prerequisite
* Activating crit and fumble rolls via `rollConfig.js`
* Using module "Dice so Nice" for dice animations

## Description
When rolling a crit or fumble no dice animation is shown but instead the result instantly appears in the chat

## Wished Behaviour
Same as when not rolling a crit/fumble: Wait for the dice animation to finish and then show the result in the chat.

## Note
I already took a look into the code. Probably the wished behaviour is not so easily possible. Technically, on a crit/fumble it makes several rolls (which would mean, multiple dice animations for one roll).

## Acceptable Bahaviour
Make one roll and wait for the result for the animation to finish, before showing it in the chat. The roll for crit/fumble doesn't need a animation for me. The main thing (for me) is that the players don't know in advance/instantly that they rolled a crit/fumble.

**Comment by pedroaugustobt** [comment]
> ## Prerequisite
> * Activating crit and fumble rolls via `rollConfig.js`
> * Using module "Dice so Nice" for dice animations
> 
> ## Description
> When rolling a crit or fumble no dice animation is shown but instead the result instantly appears in the chat
> 
> ## Wished Behaviour
> Same as when not rolling a crit/fumble: Wait for the dice animation to finish and then show the result in the chat.
> 
> ## Note
> I already took a look into the code. Probably the wished behaviour is not so easily possible. Technically, on a crit/fumble it makes several rolls (which would mean, multiple dice animations for one roll).
> 
> ## Acceptable Bahaviour
> Make one roll and wait for the result for the animation to finish, before showing it in the chat. The roll for crit/fumble doesn't need a animation for me. The main thing (for me) is that the players don't know in advance/instantly that they rolled a crit/fumble.

Were you able to fix this error? I'm analyzing the code, I came across this a little while ago

**Comment by ortegamarcel** [comment]
Yes and no. I rewrote the roll-dice logic but its not included in the system, since it is a more or less large change that could affect other places (and I didn't test it yet). But it is used by my Token Action HUD module (if enabled in the settings).
